libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -71,6 +71,8 @@ assembly::Statement Parser::parseStatement()
 		expectToken(Token::Colon);
 		assignment.variableName.location = location();
 		assignment.variableName.name = m_scanner->currentLiteral();
+		if (instructions().count(assignment.variableName.name))
+			fatalParserError("Identifier expected.");
```

Comment:
How about going for "Identifier expected, but got instruction name." and using it in the other assignment case (but in declaration).